automation_tools/satellite6/upgrade/client.py

```diff
@@ -0,0 +1,115 @@
+    """Helper function to run upgrade on docker containers as clients
+
+    :param string old_repo: The old tools repo to disable before updating
+        katello-agent package
+    :param list container_ids: The list of container_ids onto which
+        katello-agent package will be updated
+    """
+    for client in container_ids:
+        print('Upgrading client on docker container: {}'.format(client))
+        print('Previous katello-agent version:')
+        docker_execute_command(client, 'rpm -qa | grep katello-agent')
+        print('Disabling existing client tools repo:')
+        docker_execute_command(
+            client, 'subscription-manager repos --disable {}'.format(old_repo))
+        print('Upgrading clients on Docker Containers')
+        docker_execute_command(client, 'yum update katello-agent')
+        print('katello-agent package after installation:')
+        docker_execute_command(client, 'rpm -qa | grep katello-agent')
+
+
+def satellite6_client_setup():
+    """Sets up required things on upgrade running machine and on Client to
+    perform client upgrade later
+
+    If not personal clients, then it creates docker containers as clients on
+    rhevm vm.
+    """
+    # If Personal Client Hostname provided
+    if os.environ.get('CLIENT6_HOSTS') or os.environ.get('CLIENT7_HOSTS'):
+        clients6 = list(os.environ.get('CLIENT6_HOSTS'))
+        clients7 = list(os.environ.get('CLIENT7_HOSTS'))
+        # Sync latest sat tools repo to clients if downstream
+        if all([
+            os.environ.get('TOOLS_URL_RHEL6'),
+            os.environ.get('TOOLS_URL_RHEL7')
+        ]):
+            sync_tools_repos_to_upgrade('rhel6', clients6)
+            sync_tools_repos_to_upgrade('rhel7', clients7)
+    # Run upgrade on Docker Containers
+    else:
+        docker_vm = os.environ.get('DOCKER_VM')
+        clients_count = os.environ.get('CLIENTS_COUNT')
+        # First delete if any containers running
+        remove_all_docker_containers(only_running=False)
+        # Generate Clients on RHEL 7 and RHEL 6
+        clients6 = execute(generate_satellite_docker_clients_on_rhevm(
+            'rhel6', int(clients_count)/2), host=docker_vm)[docker_vm]
+        clients7 = execute(generate_satellite_docker_clients_on_rhevm(
+            'rhel7', int(clients_count)/2), host=docker_vm)[docker_vm]
+        # Sync latest sat tools repo to clients if downstream
+        if all([
+            os.environ.get('TOOLS_URL_RHEL6'),
+            os.environ.get('TOOLS_URL_RHEL7')
+        ]):
+            sync_tools_repos_to_upgrade('rhel6', clients7.keys())
+            sync_tools_repos_to_upgrade('rhel7', clients6.keys())
+        # All clients
+        clients = clients6 + clients7
+        # Refresh subscriptions on clients
+        execute(refresh_subscriptions_on_docker_clients(
+            clients.values()), host=docker_vm)
+    return clients6, clients7
+
+
+def satellite6_client_upgrade(os_version, clients):
```

Comment:
Actually I would have around 1000's of clients and its very bad(time consuming) to go in each client and get its os_version. Also this os_version is been used to pick right TOOLS_URL. So I would like to stick with the current approach as anyhow we are deciding which rhel version of client is to upgrade.